website/src/controllers/api-v3/user.js

```diff
@@ -31,4 +37,38 @@ api.getUser = {
   },
 };
 
+/**
+ * @api {post} /user/update-email
+ * @apiVersion 3.0.0
+ * @apiName EmailUpdate
+ * @apiGroup User
+ *
+ * @apiSuccess {Object} { status: 'ok' }
+ **/
+api.updateEmail = {
+  method: 'POST',
+  middlewares: [authWithHeaders(), cron],
+  url: '/user/update-email',
+  async handler (req, res) {
+    let user = res.locals.user;
+
+    if (!user.auth.local.email) throw new PreconditionFailed(res.t('userHasNoLocalRegistration'));
+
+    req.checkBody('newEmail', res.t('newEmailRequired')).notEmpty().isEmail();
+    req.checkBody('password', res.t('missingPassword')).notEmpty();
+    let validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    // check password
+    let candidatePassword = passwordUtils.encrypt(req.body.password, user.auth.local.salt);
+    if (candidatePassword !== user.auth.local.hashed_password) throw new NotAuthorized(res.t('wrongPassword'));
+
+    // save new email
+    user.auth.local.email = req.body.newEmail;
+    await user.save();
+
+    return res.respond(200, { status: 'ok', email: user.auth.local.email });
```

Comment:
@piousbox can you remove the `status: ok` part? We have discussed adding a success: false|true property but if we add it it'll be done automatically